pyte/screens.py

```diff
@@ -413,7 +416,9 @@ def reset_mode(self, *modes, **kwargs):
 
         # Lines below follow the logic in :meth:`set_mode`.
         if mo.DECCOLM in modes:
-            self.resize(columns=80)
+            if self.columns == 132 and self.saved_columns is not None:
```

Comment:
Are you sure this check is necessary? It looks like if `DECCOLM` is in `modes` then it was added there by `set_mode` and we know that the condition is automatically satisfied.

Reply:
Do you mean the `self.columns == 132` or `self.saved_columns is not None`?

The reason for checking `self.columns == 132` is to only reset to 80 characters if the terminal width has not been modified in the meantime.

The reason for `self.saved_columns is not None` is that `reset` is sending the control sequence resetting DECCOLM even when DECCOLM was not set before (and therefore DECCOLM would not be in `self.mode`, which is what I assume you meant).

Reply:
You're right, I missed that the DECOLM can indeed be reset w/o setting and it shouldn't fail the app.